process_manager/manager.py

```diff
@@ -23,6 +31,30 @@ def schedule_with_runner(self, task: SchedulerTask, mode: TaskType):
             return self.standard_runner.schedule(Process(target=task.target), task.timeout)
         else:
             raise ValueError(f'Invalid mode {mode}')
+
+    def add_task(self, start: datetime, target: callable, mode: TaskType) -> NoReturn:
+        task = SchedulerTask(start,
+                             target,
+                             self.timeout)
+        self.schedule_with_runner(task, mode)
+    
+    async def run(self, scheduler: Scheduler, period: int, mode: TaskType):
+        done = asyncio.Event()
+        
+        def shutdown():
+            done.set()
+            self.shutdown()
+            asyncio.get_event_loop().stop()
+        
+        asyncio.get_event_loop().add_signal_handler(signal.SIGINT, shutdown)
+
+        while not done.is_set():
+            self.add_task(datetime.now(), scheduler, mode)
+            if period == 0:
+                # random case #
+                await asyncio.sleep(randint(1, 10))
+            else:
+                await asyncio.sleep(period)
```

Comment:
Do we still need this random sleeping? I thought that was only for demo purposes?

Reply:
Right now is looping from task to task.  It should be deleted in the future work